src/main/java/com/andrewrs/sps/utils/Day.java

```diff
@@ -0,0 +1,99 @@
+package com.andrewrs.sps.util;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.ArrayList;
+import com.andrewrs.sps.utils.TimeRange;
+
+public class Day {
+    //Number of slots the day will be brocken into
+    private int slotCount;
+    //24 hours * 2 = 48 half hours, can be broken down further if you like, multiply by 4 instead of 2 
+    //to make it 15 minute blocks, or 12 to make it 5 minute blocks
+    private boolean[] timeSlots;
+    //incrementSize is a calculated value that is the amount of minutes of the day each block represents
+    private int incrementSize;
+    
+    public Day()
+    {
+        resize(15);
+        only9to5();
+    }
+    //this constructor takes the size of the smallest block of time allocatable in minutes, so if you
```

Comment:
Commenting is great! I think I mentioned this elsewhere, but for future projects/if you have the time, definitely recommend using javadoc for commenting.